store/sqlstore/command_store.go

```diff
@@ -134,20 +134,19 @@ func (s SqlCommandStore) PermanentDeleteByUser(userId string) store.StoreChannel
 	})
 }
 
-func (s SqlCommandStore) Update(cmd *model.Command) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		cmd.UpdateAt = model.GetMillis()
+func (s SqlCommandStore) Update(cmd *model.Command) (*model.Command, *model.AppError) {
+	cmd.UpdateAt = model.GetMillis()
 
-		if result.Err = cmd.IsValid(); result.Err != nil {
-			return
-		}
+	if err := cmd.IsValid(); err != nil {
+		return nil, err
+	}
 
-		if _, err := s.GetMaster().Update(cmd); err != nil {
-			result.Err = model.NewAppError("SqlCommandStore.Update", "store.sql_command.save.update.app_error", nil, "id="+cmd.Id+", "+err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = cmd
-		}
-	})
+	_, err := s.GetMaster().Update(cmd)
```

Comment:
nit: In this cases we discard the success and handle only the error, it's ok to do it in one if line.

Reply:
s.GetMaster().Update(cmd) returns `err` of `type error`. But we need `*model.AppError`.

Reply:
Yes, same that in the other PR, what I meant here was `if _, err := s.GetMaster().Update(cmd); err != nil {`